### PR TITLE
Using go's builtin UserConfigDirectory

### PR DESCRIPTION
**Pull request by bresilla**
This small patch searches firstly for user's config directory, e.g. in
linnux that is XDG_CONFIG_DIR and if not fond moves to placing config
files into $HOME directory. I believe this is important to not clutter
the $HOME directory.

### CODE REVIEW

File: cli/cmd/root.go [hunk 1]
```diff
@@ -111,19 +111,25 @@ func errorHandling() {
...
+    }
+
+	// Search cheat folder in home or config directory and file name "cheatrc"
+    viper.AddConfigPath(configrc)
+	viper.SetConfigName("cheatrc")
 
 	// Fallback to "vi" for the editor
 	viper.SetDefault("editor", utils.GetEnv("EDITOR", "vi"))
-	viper.SetDefault("database", "~/.cheetsheet.db")
+	viper.SetDefault("database", "~/.cheatsheet.db")
```
Comment (darrikonn) [inline cli/cmd/root.go]: Can you rebase against `master`, this has been fixed on the master branch.

**Comment by darrikonn** [comment]
I agree! Good catch!
However, I don't entirely agree with the implementation of `os.UserConfigDir`, mainly when it comes to `Darwin` and cli application. The UserConfigDir for Darwin is suitable for GUI applications.

My proposal is this.
1. Add an environment variable called `CHEAT_CONFIG`.
2. If not found, we can check the runtime. For `unix` let's fall back to `XDG_CONFIG_HOME`. Otherwise, let's use the `$HOME` directory.

Your thoughts on this?
(+ not so sure about the name `CHEAT_CONFIG`, any ideas for a better suited name?)

**Comment by bresilla** [comment]
> My proposal is this.
    Add an environment variable called CHEAT_CONFIG.
    If not found, we can check the runtime. For unix let's fall back to XDG_CONFIG_HOME. Otherwise, let's use the $HOME directory.
Your thoughts on this?
(+ not so sure about the name CHEAT_CONFIG, any ideas for a better suited name?)

I am not familiar with `darwin`but i agree with your proposal.
And `CHEAT_CONFIG` actually looks good to me too. 

Everything looks good actually. I started using this tool and I like it quite a lot actually. Will be using and reporting (and sometime contributing too).

**Comment by darrikonn** [comment]
Alright awesome, happy to hear. 
Do you want to take this forward (earning source code contribution 😁), or should I take over?

**Comment by darrikonn** [comment]
Implemented in https://github.com/darrikonn/cheat/pull/4

Version `1.1.1` contains the fix/feature.
Thanks for reporting :)!

**Pull request closed without merging**